lib/geo.js

```diff
@@ -12,19 +12,31 @@ var assert = require('assert');
 exports.nearFilter = function nearFilter(where) {
   var result = false;
 
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
+  var keyPath = [];
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      var keys = Object.keys(node);
+
+      for (var i = 0; i < keys.length; i++) {
```

Comment:
I ended up using a `for` loop because there's no easy way to break out of a `forEach` in JavaScript (see LNs [33-35](https://github.com/strongloop/loopback-datasource-juggler/pull/1028/files/5c090feeb4a2917b80e6b146efbeb88876a41940#diff-b367a815d81f15531e0d0553ea8aaa5aR33-35))... also http://stackoverflow.com/a/2641374/2305837 - I figured there's no reason to keep recursing once the `near` filter is found.